src/termcolor/termcolor.py

```diff
@@ -120,11 +121,19 @@ def _can_do_colour(
         return False
     if "FORCE_COLOR" in os.environ:
         return True
-    return (
-        hasattr(sys.stdout, "isatty")
-        and sys.stdout.isatty()
-        and os.environ.get("TERM") != "dumb"
-    )
+
+    try:
+        return (
+            hasattr(sys.stdout, "fileno")
+            and os.isatty(sys.stdout.fileno())
+            and os.environ.get("TERM") != "dumb"
+        )
+    except io.UnsupportedOperation:
+        return (
+            hasattr(sys.stdout, "isatty")
+            and sys.stdout.isatty()
+            and os.environ.get("TERM") != "dumb"
+        )
```

Comment:
Perhaps something like this to (a) move the duplicate checks out of the `try`/`except` and (b) return early after fewer checks?

```suggestion
    # Then check system:
    if os.environ.get("TERM") == "dumb":
        return False
    if not hasattr(sys.stdout, "fileno"):
        return False

    try:
        return os.isatty(sys.stdout.fileno())
    except io.UnsupportedOperation:
        return sys.stdout.isatty()
```

And is it possible to add a test case to cover the `except` block?

Reply:
This look much better, less duplicate checks